src/options.js

```diff
@@ -0,0 +1,119 @@
+/*global chrome */
+function restore_options() {
+    loadDoNotReplaceList(function(doNotReplaceList) {
+        displayDoNotReplaceList(doNotReplaceList);
+    });
+}
+
+function loadDoNotReplaceList(callback) {
+    chrome.storage.sync.get(
+        {
+            doNotReplaceList: []
+        },
+        function(items) {
+            if (callback) {
+                callback(items.doNotReplaceList);
+            }
+        }
+    );
+}
+
+function saveDoNotReplaceList(doNotReplaceList, callback) {
+    const items = { doNotReplaceList: doNotReplaceList };
+    chrome.storage.sync.set(items, function(items) {
+        if (callback) {
+            callback();
+        }
+    });
+}
+
+function addItem() {
+    const newEntryField = document.getElementById("newEntry");
+
+    let url = newEntryField.value;
+
+    if (url.includes("://")) {
+        // Strip off the protocol using the first instance of '://'
+        url = url
+            .split("://")
+            .slice(1)
+            .join("://");
+    }
+
+    url = url.split("/")[0];
+
+    newEntryField.value = "";
+    loadDoNotReplaceList(function(doNotReplaceList) {
+        doNotReplaceList.push(url);
+        doNotReplaceList.sort();
+        saveDoNotReplaceList(doNotReplaceList);
+
+        // Update the displayed list
+        displayDoNotReplaceList(doNotReplaceList);
+    });
+}
+
+function removeItem(urlToRemove) {
+    loadDoNotReplaceList(function(doNotReplaceList) {
+        doNotReplaceList.sort();
+        const index = doNotReplaceList.findIndex(function(s) {
+            return s === urlToRemove;
+        });
+
+        // If the item hasn't already been removed from the list,
+        if (index !== -1) {
+            doNotReplaceList.splice(index, 1);
+        } else {
+            console.log(urlToRemove + " was not found in `doNotReplaceList`");
+        }
+
+        saveDoNotReplaceList(doNotReplaceList);
+
+        // Update the displayed list
+        displayDoNotReplaceList(doNotReplaceList);
```

Comment:
Putting `saveDoNotReplaceList` in the conditional makes sense; we don't need to overwrite the list if we haven't changed it. We'll still update the displayed list either way because we want to keep it synced with the list in `chrome.storage` whenever possible.